.gitpod.yml

```diff
@@ -1,40 +1,54 @@
 github:
   prebuilds:
-    # enable for the master/default branch (defaults to true)
     master: true
-    # enable for all branches in this repo (defaults to false)
     branches: true
-    # enable for pull requests coming from this repo (defaults to true)
     pullRequests: true
-    # enable for pull requests coming from forks (defaults to false)
     pullRequestsFromForks: true
-    # add a check to pull requests (defaults to true)
     addCheck: false
-    # add a "Review in Gitpod" button as a comment to pull requests (defaults to false)
     addComment: false
-    # add a "Review in Gitpod" button to the pull request's description (defaults to false)
     addBadge: false
-    # add a label once the prebuild is ready to pull requests (defaults to false)
     addLabel: false
 tasks:
-  - name: dev-mode
+  - name: setup
     init: |
-      pip3 install -e ".[docs,test]"
-      jlpm install
-      jlpm run build
+      pushd /workspace
+      wget -qO- https://micro.mamba.pm/api/micromamba/linux-64/latest | tar -xvj bin/micromamba
+      popd
+      # bootstrap activation commands for other tasks to reuse
+      cat <<EOT > /workspace/bin/activate-env.sh
+      export MAMBA_ROOT_PREFIX=/workspace/.micromamba
+      export MAMBA_EXE=/workspace/bin/micromamba
+      $(/workspace/bin/micromamba shell hook --shell=bash)
+      export JUPYTER_PREFER_ENV_PATH=1
+      micromamba activate
+      EOT
+      source /workspace/bin/activate-env.sh
+      micromamba config append channels conda-forge
+      micromamba install -n base -y python=3.10 nodejs=14 yarn
+      python -m pip install -e ".[dev,docs,test]" && jlpm install && jlpm run build
       gp sync-done setup
```

Comment:
Since all these extras to use mamba are being added, I would highly recommend specifying a custom Dockerfile (see https://www.gitpod.io/docs/config-docker#configure-a-custom-dockerfile) this would help keep this config file cleaner and also removes the need to add later the `echo "source /workspace/bin/activate-env.sh" >> ~/.bashrc` command in line 40. 
Plus, we can ensure that the corresponding environment is always activated by default and reduce the need to have all the `gp sync` related commands. 
This is the approach currently used in NumPy and SciPy. This setup and leveraging prebuilds ensures that the Gitpod instance is pretty much ready to use once someone starts a new workspace.

Reply:
Right, I think the main motivation for proceeding like this was to keep things as simple as possible and self-contained in one file. It's not clear whether folks are actively using Gitpod at the moment, this didn't seem to be the case when the question was raised at a JupyterLab weekly call a few weeks ago.

Switching to a `Dockerfile` sounds good, although Docker can also be complicated and requires extra knowledge to maintain. Although it could also pave the way to support GitHub Codespaces at some point to with `devcontainer`.